pdr_backend/predictoor/util.py

```diff
@@ -40,6 +40,23 @@ def find_shared_slots(
     return result
 
 
+def count_unique_slots(shared_slots: List[Tuple[List[str], List[int]]]) -> int:
+    """
+    Calculates the number of unique slots from the output of the `find_shared_slots`.
+
+    @param shared_slots: List[Tuple[List[str], List[int]]]
+        The output of the find_shared_slots function.
+
+    @return
+    int: The number of unique slots.
+    """
+    all_slots = []
+    for _, slots in shared_slots:
```

Comment:
return len({all_slots.extend(sl) for sl in shared_slots.values()})
I think {} will be enough for it to be a set, but if not you can use the `set` syntax as well